### ISSUE TITLE
VimEnter,DirChanged instead of BufReadPre

### ISSUE THREAD

**Issue by ii14** [jenterkin/vim-autosource#4]
Wouldn't it make more sense to load configs based on actual current working directory (`getcwd()`) instead of what files are loaded?

Here is how I did it in my plugin:

```vim
  autocmd VimEnter * call s:Source()
  if exists("#DirChanged")
    autocmd DirChanged * if v:event['scope'] ==# 'global' | call s:Source() | endif
  endif
```

**Comment by jenterkin** [comment]
That's what I initially had, but I'm kind of all over the place when it comes to what I want loaded, so it didn't work out too well for how I work.

For example, I'm working on a web application that has a python backend and a bunch of typescript for the frontend. I have two separate confs for those since I don't really want to configure a python language server when I'm working on typescript and vice-versa.

So one example where this doesn't work when based off of cwd is if I'm writing some api call on the frontend, realize I need to make a quick edit to the response server-side, so I do a quick `:vsplit ~/path/to/python/code` because the config for that likely won't be in my cwd.

So tl;dr it kind of just loads what you need as you need it.

I can certainly imagine scenarios where this can clutter the conf of your current session though. Maybe it would make sense to make this a setting? e.g. `g:autosource_by_cwd = 1`

Also https://github.com/jenterkin/vim-autosource/issues/7 should support this via custom autocmds.

**Comment by ii14** [comment]
Without the `v:event['scope'] ==# 'global'` check you could use `:lcd` for window or `:tcd` for tab.

**Comment by jenterkin** [comment]
Oh yeah, this isn't really a problem for me since I set `autochdir`, I just more meant that I don't know that I want to have users have to set that or similar options to achieve this kind of flow, since setting those options can have wider effects.

My current thoughts are that most users would expect this plugin to source from `$HOME` to the file they are editing (though there's unfortunately no way of validating this hunch right now), so I think the current functionality makes sense as a sane default, at least for now.

I'll add in the `g:autosource_by_cwd` option and put it in the readme so that it's easily configurable without users having to write their own `autocmd`s.

**Comment by jenterkin** [comment]
I've given this a bit more thought. I think I'm going to hold off on adding an option for this since support for disabling the `autocmd` and creating custom `autocmd`s landed. I think what exists is a good default, but will definitely change if there's a significant number of users with a use-case where it doesn't work.
